File: geo-project/geo-server/src/main/java/com/huanjing/geo/module/content/constant/ArticleTypes.java
package com.huanjing.geo.module.content.constant;

import java.util.Map;
import java.util.Set;

public final class ArticleTypes {
    public static final String GENERAL_ARTICLE = "general_article";
    public static final String FAQ = "faq";
    public static final String SCENARIO_CONTENT = "scenario_content";
    public static final String INDUSTRY_ARTICLE = "industry_article";
    public static final String STAGE_ADVICE = "stage_advice";
    public static final String BUYING_GUIDE = "buying_guide";
    public static final String COMPARISON = "comparison";
    public static final String COST_ANALYSIS = "cost_analysis";
    public static final String PITFALL_GUIDE = "pitfall_guide";
    public static final String SOCIAL_NOTE = "social_note";
    public static final String NEWS_BRIEF = "news_brief";
    public static final String FORUM_DISCUSSION = "forum_discussion";

    public static final Map<String, String> LABELS = Map.ofEntries(
            Map.entry(GENERAL_ARTICLE, "通用文章"),
            Map.entry(FAQ, "问答文章"),
            Map.entry(SCENARIO_CONTENT, "场景内容"),
            Map.entry(INDUSTRY_ARTICLE, "行业文章"),
            Map.entry(STAGE_ADVICE, "阶段建议"),
            Map.entry(BUYING_GUIDE, "选择指南"),
            Map.entry(COMPARISON, "对比评测"),
            Map.entry(COST_ANALYSIS, "费用解析"),
            Map.entry(PITFALL_GUIDE, "避坑指南"),
            Map.entry(SOCIAL_NOTE, "经验笔记"),
            Map.entry(NEWS_BRIEF, "资讯简讯"),
            Map.entry(FORUM_DISCUSSION, "讨论帖")
    );

    private static final Set<String> SUPPORTED_TYPES = Set.copyOf(LABELS.keySet());

    private static final Set<String> GENERATED_CONTENT_TYPES = Set.of(
            FAQ,
            SCENARIO_CONTENT,
            INDUSTRY_ARTICLE,
            STAGE_ADVICE,
            BUYING_GUIDE,
            COMPARISON,
            COST_ANALYSIS,
            PITFALL_GUIDE,
            SOCIAL_NOTE,
            NEWS_BRIEF,
            FORUM_DISCUSSION
    );

    private ArticleTypes() {
    }

    public static boolean isSupported(String articleType) {
        return SUPPORTED_TYPES.contains(articleType);
    }

    public static boolean isGeneratedContentType(String articleType) {
        return GENERATED_CONTENT_TYPES.contains(articleType);
    }

    public static String label(String articleType) {
        if (articleType == null) {
            return null;
        }
        return LABELS.getOrDefault(articleType, articleType);
    }
}

File: geo-project/geo-web/src/constants/articleTypes.ts
export const ARTICLE_TYPE_OPTIONS = [
  { value: 'general_article', label: '通用文章' },
  { value: 'faq', label: '问答文章' },
  { value: 'scenario_content', label: '场景内容' },
  { value: 'industry_article', label: '行业文章' },
  { value: 'stage_advice', label: '阶段建议' },
  { value: 'buying_guide', label: '选择指南' },
  { value: 'comparison', label: '对比评测' },
  { value: 'cost_analysis', label: '费用解析' },
  { value: 'pitfall_guide', label: '避坑指南' },
  { value: 'social_note', label: '经验笔记' },
  { value: 'news_brief', label: '资讯简讯' },
  { value: 'forum_discussion', label: '讨论帖' },
] as const

export const GENERATED_ARTICLE_TYPE_OPTIONS = ARTICLE_TYPE_OPTIONS.filter((item) => item.value !== 'general_article')

const ARTICLE_TYPE_LABELS: Readonly<Record<string, string>> = Object.fromEntries(
  ARTICLE_TYPE_OPTIONS.map((item) => [item.value, item.label]),
)

export function articleTypeLabel(value?: string | null): string {
  return value ? ARTICLE_TYPE_LABELS[value] || value : '-'
}
